symcrypt-sys/build/static_link.rs

```diff
@@ -0,0 +1,422 @@
+use super::triple::Triple;
+use std::collections::HashSet;
+
+const LIB_NAME: &str = "symcrypt";
+
+/// Compiles and links the SymCrypt library statically.
+/// This is the entery point for building SymCrypt statically.
+///
+/// - Determines the build configuration based on target architecture.
+/// - Calls `compile_symcrypt_static()` to actually build SymCrypt.
+/// - Outputs necessary metadata for Cargo (`cargo:rustc-link-lib=...`).
+///
+/// Based on SymCrypt's `CMakeLists.txt`, but adapted for Rust.
+pub fn compile_and_link_symcrypt() -> std::io::Result<()> {
+    let options = SymCryptOptions::new();
+    println!("Build config: {:?}", options);
+
+    // Required Windows bcrypt dependency for BCryptGenRandom
+    const ADDITIONAL_DEPENDENCIES: &[&str] = &[
+        #[cfg(windows)]
+        "bcrypt",
+    ];
+    println!("cargo:rerun-if-changed=upstream");
+    println!("Compiling SymCrypt...");
+
+    // Compile and Build SymCrypt with provided SymCryptOptions
+    compile_symcrypt_static(LIB_NAME, &options)?;
+    println!("cargo:rustc-link-lib=static={LIB_NAME}");
+
+    // Link additional dependencies
+    for dep in ADDITIONAL_DEPENDENCIES {
+        println!("cargo:rustc-link-lib=dylib={dep}");
+    }
+
+    Ok(())
+}
+
+// TODO: update -symcrypt_fips_build comment
+
+/// Holds configuration options for compiling SymCrypt.
+///
+/// - `triple`: The target triple
+/// - `symcrypt_use_asm`: Whether to enable assembly optimizations.
+/// - `symcrypt_fips_build`: Whether to build in FIPS mode PLACEHOLDER
+/// - `preconfiged_cc`: Returns a Pre-configured `cc` object for the target triple.
+#[derive(Debug)]
+struct SymCryptOptions {
+    triple: Triple,
+    symcrypt_use_asm: bool,
+    //symcrypt_fips_build: bool, // TODO: Determine if we should expose FIPS build option?
+}
+
+impl SymCryptOptions {
+    fn new() -> Self {
+        Self {
+            triple: Triple::get_target_triple(),
+            symcrypt_use_asm: false, // FIXME: Turn this to true when we get ASM checked in
```

Comment:
As mentioned in a meeting before we don't want to explicitly say that a certain build is not FIPS compliant, but in this scenario I think it could make sense alerting customers that static linking is strictly not FIPS.